application/managers/applications.py

```diff
@@ -269,18 +274,38 @@ async def get_component_health(self, component: Component, application: Applicat
         """
         Returns component health condition.
         """
-        try:
-            component_health_status = await self.helm_manager.release_health_status(
-                application.organization,
-                application.context_name,
-                application.namespace,
-                component.name
-            )
-            health_status = component_health_status['status']
-            details = component_health_status['details']
-        except ReleaseNotFoundException:
-            health_status = ReleaseHealthStatuses.unhealthy
-            details = {}
+        match component.type:
+            case ComponentTypes.helm_chart:
+                try:
+                    component_health_status = await self.helm_manager.release_health_status(
+                        application.organization,
+                        application.context_name,
+                        application.namespace,
+                        component.name
+                    )
+                    health_status = component_health_status['status']
+                    details = component_health_status['details']
+                except ReleaseNotFoundException:
+                    health_status = ReleaseHealthStatuses.unhealthy
+                    details = {}
+
+            case ComponentTypes.http:
+                try:
+                    component_health_status = await self.http_manager.http_request(
+                        organization=application.organization,
+                        context_name=application.context_name,
+                        component_name=component.name,
+                        url=component.status.url,
+                        method=component.status.method,
+                        headers=component.status.headers,
+                        parameters=component.status.parameters,
+                        dry_run=False
+                    )
+                    health_status = ReleaseHealthStatuses.healthy
+                    details = component_health_status
+                except Exception as error:
```

Comment:
Please never try to catch all exceptions. Always catch only what you want to catch. Or at least propagate exception further by calling `raise` after 308 line so other part of project can be averred about occurred exceptions.